Tests/FluentAssertions.Equivalency.Specs/DataRowSpecs.cs

```diff
@@ -1,332 +1,421 @@
-                var dataRow2 = dataSet2.TypedDataTable1[0];
+        [Fact]
+        public void When_data_does_not_match_and_column_is_not_excluded_it_should_fail()
+        {
+            // Arrange
+            var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
 
-                dataRow2.RowError = "Manually added error";
+            var dataSet2 = new TypedDataSetSubclass(dataSet1);
 
-                // Act
-                Action action = () => dataRow1.Should().BeEquivalentTo(dataRow2);
+            var dataRow1 = dataSet1.TypedDataTable1[0];
+            var dataRow2 = dataSet2.TypedDataTable1[0];
 
-                // Assert
-                action.Should().Throw<XunitException>().Which.Message.Should().Contain("HasErrors");
-            }
+            dataRow2.String = Guid.NewGuid().ToString();
+            dataSet2.AcceptChanges();
 
-            [Fact]
-            public void When_HasErrors_does_not_match_and_property_is_excluded_it_should_succeed()
-            {
-                // Arrange
-                var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+            // Act
+            Action action = () => dataRow1.Should().BeEquivalentTo(dataRow2);
 
-                var dataSet2 = new TypedDataSetSubclass(dataSet1);
+            // Assert
+            action.Should().Throw<XunitException>().Which.Message.Should().Contain(dataRow2.String);
+        }
 
-                var dataRow1 = dataSet1.TypedDataTable1[0];
-                var dataRow2 = dataSet2.TypedDataTable1[0];
+        [Fact]
+        public void When_data_does_not_match_and_column_is_excluded_it_should_succeed()
+        {
+            // Arrange
+            var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
 
-                dataRow2.RowError = "Manually added error";
+            var dataSet2 = new TypedDataSetSubclass(dataSet1);
 
-                // Act & Assert
-                dataRow1.Should().BeEquivalentTo(dataRow2, config => config.Excluding(dataRow => dataRow.HasErrors));
-            }
+            var dataRow1 = dataSet1.TypedDataTable1[0];
+            var dataRow2 = dataSet2.TypedDataTable1[0];
 
-            [Fact]
-            public void When_RowState_does_not_match_and_property_is_not_excluded_it_should_fail()
-            {
-                // Arrange
-                var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+            dataRow2.DateTime = DateTime.UtcNow;
+            dataSet2.AcceptChanges();
 
-                var dataSet2 = new TypedDataSetSubclass(dataSet1);
+            // Act & Assert
+            dataRow1.Should().BeEquivalentTo(dataRow2, config => config.ExcludingColumn(dataSet2.TypedDataTable1.DateTimeColumn));
+        }
 
-                var dataRow1 = dataSet1.TypedDataTable1[0];
-                var dataRow2 = dataSet2.TypedDataTable1[0];
+        [Fact]
+        public void When_data_has_column_asserting_HaveColumn_should_succeed()
```

Comment:
❌ Keep the names functional and concise without using the literal names of code elements. Same for all the other tests.